go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -261,12 +278,47 @@ func getDbStates(ctx context.Context, mrEnv env.MultiRepoEnv, dbs []dsqle.Databa
 		}
 
 		head := dEnv.RepoStateReader().CWBHeadSpec()
```

Comment:
Would replace this with the definition from repo state:

head, _ := doltdb.NewCommitSpec("HEAD")

Makes it more obvious what's going on here. The CWBHeadSpec() method is pretty dumb, needs to go

Reply:
Also the conditional above (dEnv == nil) is dead code